src/app/features/room/RoomInput.tsx

```diff
@@ -289,25 +288,41 @@ export const RoomInput = forwardRef<HTMLDivElement, RoomInputProps>(
 
       if (plainText === '') return;
 
-      let body = plainText;
-      let formattedBody = customHtml;
-      if (replyDraft) {
-        body = parseReplyBody(replyDraft.userId, trimReplyFromBody(replyDraft.body)) + body;
-        formattedBody =
-          parseReplyFormattedBody(
-            roomId,
-            replyDraft.userId,
-            replyDraft.eventId,
-            replyDraft.formattedBody
-              ? trimReplyFromFormattedBody(replyDraft.formattedBody)
-              : sanitizeText(replyDraft.body)
-          ) + formattedBody;
-      }
+      const body = plainText;
+      const formattedBody = customHtml;
 
       const content: IContent = {
         msgtype: msgType,
         body,
       };
+      const userIdMentions = new Set<string>();
+      if (replyDraft) {
+        userIdMentions.add(replyDraft.userId);
+      }
+      let mentionsRoom = false;
+      editor.children.forEach((node: Descendant): void => {
+        if("type" in node && node.type !== "paragraph") return;
+        const paragraph: ParagraphElement = node as ParagraphElement;
```

Comment:
the if condition can be changed to:
```js
if (!("type" in node) || node.type !== BlockType.Paragraph) return;
```
OR simply
```js
if ("type" in node || node.type === BlockType.Paragraph) {/*....do paragraph stuff*/}
```

as currently `Text` can be interpreted as `ParagraphElement` when `"type" in node` turn out to be false.